action/protocol/staking/candidate_buckets_indexer_test.go

```diff
@@ -54,49 +52,35 @@ func TestCandidatesBucketsIndexer_PutGetCandidates(t *testing.T) {
 		require.NoError(cbi.PutCandidates(v.height, v.candidates))
 	}
 
-	//incorrect parameter
 	tests2 := []struct {
 		height        uint64
 		offset, limit uint32
+		r1            int
+		r2            uint64
 	}{
-		{0, 0, 0},
-		{2, 1111111, 1},
-		{21111111, 1111111, 1333333},
+		{0, 0, 0, 0, 0},
+		{2, 0, 1, 1, 2},
+		{2, 0, 100, 1, 2},
+		{2, 1234, 0, 0, 2},
+		{2, 1111111, 1, 0, 2},
+		{21111111, 1111111, 1333333, 0, 2},
 	}
 	for _, v2 := range tests2 {
-		_, _, err = cbi.GetCandidates(v2.height, v2.offset, v2.limit)
+		a, b, err := cbi.GetCandidates(v2.height, v2.offset, v2.limit)
+		t.Logf("%v, a=%s, b= %d", v2, a, b)
 		require.NoError(err)
-	}
 
-	b, h, err := cbi.GetCandidates(tests[1].height, 0, 1)
-	require.NoError(err)
-	require.Equal(h, tests[1].height)
-
-	var r iotextypes.CandidateListV2
-	require.NoError(proto.Unmarshal(b, &r))
-	require.Equal(len(r.Candidates), 1)
-	require.Equal(r.Candidates[0].Name, tests[1].candidates.Candidates[0].Name)
-	require.Equal(r.Candidates[0].SelfStakeBucketIdx, tests[1].candidates.Candidates[0].SelfStakeBucketIdx)
+		var r iotextypes.CandidateListV2
+		require.NoError(proto.Unmarshal(a, &r))
```

Comment:
can you do the same for PutGetBuckets(t *testing.T)?

including the if() check below